ParquetClassLibrary/Sandbox/Room.cs

```diff
@@ -0,0 +1,128 @@
+        /// The <see cref="Space"/>s whose <see cref="Block"/>s and <see cref="Furnishing"/>s
+        /// define the limits of this <see cref="Room"/>.
+        /// </summary>
+        public readonly HashSet<Space> Perimeter;
+
+        /// <summary>
+        /// The cached <see cref="EntityID"/>s for every <see cref="Furnishing"/> found in this <see cref="Room"/>
+        /// together with the number of times that furnishing occurs.
+        /// </summary>
+        private Dictionary<EntityID, int> _cachedFurnishings;
+
+        /// <summary>
+        /// The <see cref="EntityID"/>s for every <see cref="Furnishing"/> found in this <see cref="Room"/>
+        /// together with the number of times that furnishing occurs.
+        /// </summary>
+        public Dictionary<EntityID, int> Furnishings
+        {
+            get
+            {
+                return _cachedFurnishings
+                       ?? (_cachedFurnishings = WalkableArea
+                                                .Concat(Perimeter)
+                                                .Where(space => null != space.Content.Furnishing
+                                                             && space.Content.Furnishing.ID != EntityID.None)
+                                                .GroupBy(space => space.Content.Furnishing.ID)
+                                                .ToDictionary(group => group.Key, group => group.Count()));
+            }
+        }
+
+        /// <summary>
+        /// The location with the least X and Y coordinates of every <see cref="Space"/> in this <see cref="Room"/>.
+        /// </summary>
+        /// <remarks>
+        /// A value of <c>null</c> indicates that this <see cref="Room"/> has yet to be located.
+        /// </remarks>
+        private Vector2Int? _cachedPosition;
+
+        /// <summary>
+        /// A location with the least X and Y coordinates of every <see cref="Space"/> in this <see cref="Room"/>.
+        /// </summary>
+        /// <remarks>
+        /// This location could server as a the upper, left point of a bounding rectangle entirely containing the room.
+        /// </remarks>
+        public Vector2Int Position
+            => (Vector2Int)(
+                null == _cachedPosition
+                    ? _cachedPosition = new Vector2Int(Perimeter.Select(space => space.Position.X).Min(),
+                                                       Perimeter.Select(space => space.Position.X).Min())
+                    : _cachedPosition);
+
+        /// <summary>
+        /// The cached <see cref="RoomRecipe"/> identifier.
+        /// A value of <see cref="EntityID.None"/> indicates that this <see cref="Room"/>
+        /// has yet to be matched with a <see cref="RoomRecipe"/>.
+        /// </summary>
+        private EntityID _cachedRecipeID = EntityID.None;
+
+        /// <summary>The <see cref="RoomRecipe"/> that this <see cref="Room"/> matches.</summary>
+        public EntityID RecipeID
```

Comment:
Can this cache None as an answer? Looks like it might always try to look up the room's recipe if it didn't have one before. Probably not a lot of processing effort, so can be left as is.

Reply:
So, when I wrote this I was thinking that None should never be a valid answer... but of course there will be Rooms that do not correspond to any Room Recipe so I need to rethink this.  Thanks!